tests/unit/services/test_package.py

```diff
@@ -57,18 +57,24 @@ def test_pack_target_arch(default_project, default_factory, mocker):
         services=default_factory,
         platform="amd64",
         build_for="s390x",
+        snapcraft_yaml_path=tmp_path / "snapcraft.yaml",
     )
 
-    package_service.pack(prime_dir=Path("prime"), dest=Path())
+    package_service.pack(prime_dir=tmp_path / "prime", dest=tmp_path)
 
     assert mock_pack_snap.call_args.kwargs["target_arch"] == "s390x"
 
 
-def test_metadata(package_service, default_factory, default_build_plan, new_dir):
+def test_metadata(
```

Comment:
can we add a new test (new please), with setenv for the manifest to be generated?

Reply:
Done! While I was at it I ensured that the manifest wasn't generated if we didn't ask for it. I also added a unit test for `generate_manifest` on the lifecycle service that we can further parametrize as needed.